src/components/Tabs/ScatterPlotTab.tsx

```diff
@@ -76,48 +71,41 @@ export default memo(function ScatterPlotTab(props: ScatterPlotTabProps): ReactEl
+  const colorRamp = useViewerStateStore((state) => state.colorRamp);
+  const currentFrame = useViewerStateStore((state) => state.currentFrame);
+  const inRangeLUT = useViewerStateStore((state) => state.inRangeLUT);
+  const outlierDrawSettings = useViewerStateStore((state) => state.outlierDrawSettings);
+  const outOfRangeDrawSettings = useViewerStateStore((state) => state.outOfRangeDrawSettings);
+  const rangeType = useViewerStateStore((state) => state.scatterRangeType);
+  const selectedFeatureKey = useViewerStateStore((state) => state.featureKey);
+  const selectedTrack = useViewerStateStore((state) => state.track);
+  const setFrame = useViewerStateStore((state) => state.setFrame);
+  const setRangeType = useViewerStateStore((state) => state.setScatterRangeType);
+  const setTrack = useViewerStateStore((state) => state.setTrack);
+  const setXAxis = useViewerStateStore((state) => state.setScatterXAxis);
+  const setYAxis = useViewerStateStore((state) => state.setScatterYAxis);
+  const xAxisFeatureKey = useViewerStateStore((state) => state.scatterXAxis);
+  const yAxisFeatureKey = useViewerStateStore((state) => state.scatterYAxis);
 
   // Debounce changes to the dataset to prevent noticeably blocking the UI thread with a re-render.
-  const dataset = useDebounce(store.dataset, 500);
-  const categoricalPalette = useDebounce(store.categoricalPalette, 100);
-  const [colorRampMin, colorRampMax] = useDebounce(store.colorRampRange, 100);
+  const rawDataset = useViewerStateStore((state) => state.dataset);
+  const rawCategoricalPalette = useViewerStateStore((state) => state.categoricalPalette);
+  const rawColorRampRange = useViewerStateStore((state) => state.colorRampRange);
+  const dataset = useDebounce(rawDataset, 500);
+  const categoricalPalette = useDebounce(rawCategoricalPalette, 100);
+  const [colorRampMin, colorRampMax] = useDebounce(rawColorRampRange, 100);
+
+  const isDebouncePending =
+    dataset !== rawDataset || colorRampMin !== rawColorRampRange[0] || colorRampMax !== rawColorRampRange[1];
+
+  const { isPlaying, isVisible } = props;
 
-  const {
-    colorRamp,
-    inRangeLUT,
-    currentFrame,
-    featureKey: selectedFeatureKey,
-    track: selectedTrack,
-    setTrack,
-    setFrame,
-    clearTrack,
-  } = store;
-  const { isPlaying, isVisible, viewerConfig } = props;
-
-  const [isPending, startTransition] = useTransition();
-  // This might seem redundant with `isPending`, but `useTransition` only works within React's
-  // update cycle. Plotly's rendering is synchronous and can freeze the state update render,
-  // so we need to track completion with a separate flag.
   // TODO: `isRendering` sometimes doesn't trigger the loading spinner.
   const [isRendering, setIsRendering] = useState(false);
 
   const plotDivRef = React.useRef<HTMLDivElement>(null);
-  const plotRef = React.useRef<Plotly.PlotlyHTMLElement | null>(null);
+  const [plotRef, setPlotRef] = useState<Plotly.PlotlyHTMLElement | null>(null);
```

Comment:
This value was changed to state so that updates to it would trigger a re-render. When it was a ref, I sometimes saw a bug where the scatterplot could not be interacted with.